### ISSUE TITLE
Realm naming conventions/performance

### ISSUE THREAD

**Issue by Happy-Ferret** [wamp-proto/wamp-proto#323]
I'm new to WAMP and in the process of using the Go implementation ([nexus](https://github.com/gammazero/nexus)) to write my own little desktop bus (sort of like DBUS, but easier and web transparent).

So far, it all works rather well. However, I am wondering about naming conventions.
Do they apply to realms, too?  Currently, I have one realm registered as ``desktop@happy-ferret.com``.
Could there be any situation in which this would fail, and thus ``com.happy-ferret.desktop``/``desktop.happy-ferret.com`` would be preferable?

The router will occasionally log something like ``YIELD received with unknown invocation request ID: 19``. So far, this only happened with the former naming scheme, not the latter two. Coincidence or a side-effect of using ``@`` inside the realm URI?

Lastly, what would you suggest is the better way to organize procedures/messages? Is it better/more efficient to use short names and multiple realms (i e ``com.happy-ferret.screenshot``, ``com.happy-ferret.notifications``, etc for realm names and short procedure names like ``TakeScreenshot``) or one single realm and fully qualified names (i e procedures would be named ``com.happy-ferret.screenshot.TakeScreenshot``, similarly to DBUS)? Are large realms with many procedures and messages bound to slow things down?

EDIT: NVM about the ``YIELD`` part of the question. That was related to something else. Namely, not properly returning after an RPC. The rest of the question(s) still stands.

**Comment by ecorm** [comment]
What does this have to do with the WAMP protocol itself? Support requests for the nexus implementation should be sent to the appropriate channel under the nexus project.

**Comment by Jopie64** [comment]
This question was not Nexus specific. It was about naming conventions used in wamp, and hence I think a valid question for here, although I agree that it might be better to ask these things in the wamp Google group.
https://groups.google.com/forum/m/#!forum/wampws

To answer part of the question:
Although the protocol doesn't require it, a reversed fqdn convention like in Java is often implied.
I'm not sure about the realm name convention. Keep in mind though that every session can join just 1 realm. So to join multiple realms you need to connect multiple times. Hence I think it is better to use fully qualified names for your procedures/topics.

**Comment by ecorm** [comment]
Yes, the Google group is more appropriate for this question, unless the question raises an issue about information missing in the WAMP spec.

I'll let the project founders decide if this issue should be closed.

**Comment by oberstet** [comment]
@Happy-Ferret first off, welcome to the WAMP implementors and users community!

Yeah, WAMP could indeed be a better Dbus. The best place to announce your project, have a discussion etc would be the WAMP mailing list (also see below).

Rgd you specific Q:

Realm names must follow the same rules as all other identifier in WAMP (URIs) - however, this is missing in the spec text: https://wamp-proto.org/static/rfc/draft-oberstet-hybi-crossbar-wamp.html#rfc.section.5.1 

So this is the bug here: I have renamed the issue accordingly. 

All other questions raised are implementation specific or application specific (how to best structure URIs in the overall app, or if and when to use multiple realms).

As  @ecorm mentions, these questions are best posted on the WAMP mailing list or Gitter channel. Those channels are for all WAMP implementors and users of WAMP - they are "vendor neutral".

The repo here is for the WAMP protocol specification in particular. Of course there is a "gray area": as in, implementation A behaves like X, but B like Y. And we can discuss if it makes sense to standardize on either X or Y.

The general approach is: the WAMP spec is a common denominator, but implementations are free to step forward, experiment and expand on top of that. When a feature appears that survies the real world and users test, we can take up the feature for formal specification (in the advanced profile part).

* The WAMP basic profile (BP): that should be fixed and set in stone .. we are mostly in "polish/gapfill" mode - the design is closed ...
* The WAMP advanced profile (AP): that is the where new stuff rgd WAMP protocol is growing ..

Reason for a BP/AP split: strike a balance between a stable core without prohibiting further evolution.

The underlying paradigm carrying all this is the IETF mantra: rough consensus, working code.

**Comment by oberstet** [comment]
related: https://github.com/wamp-proto/wamp-proto/issues/220
